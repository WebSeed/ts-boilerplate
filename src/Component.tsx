import * as React from 'react'
import styled from 'styled-components'

const StyledComponent = styled.div`
  background-color: green;
`

const MyComponent = () => (
  <StyledComponent data-testid="component-id">Hello World</StyledComponent>
)

export default MyComponent
