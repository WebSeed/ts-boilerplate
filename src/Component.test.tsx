/* eslint-disable import/no-extraneous-dependencies */
import * as React from 'react'
import 'jest-styled-components'
import 'jest-dom/extend-expect'
import { render } from 'react-testing-library'

import Component from './Component'

describe('<Component>', () => {
  it('matches snapshot', () => {
    const { container } = render(<Component />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('has the correct content', () => {
    const { getByTestId } = render(<Component />)
    expect(getByTestId('component-id')).toHaveTextContent('Hello World')
  })
})
