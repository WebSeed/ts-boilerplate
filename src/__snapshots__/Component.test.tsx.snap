// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Component> matches snapshot 1`] = `
.c0 {
  background-color: green;
}

<div
  class="c0"
  data-testid="component-id"
>
  Hello World
</div>
`;
